File: 13-10-2024_Delete_Alternate_Nodes.cpp
// Function to delete alternate nodes in a linked list
class Solution {
  public:
    void deleteAlt(struct Node *head) {
        // If the list is empty, return
        if (head == NULL)
            return;

        // Initialize two pointers, prev and node, to keep track of current and next
        // nodes
        struct Node *prev = head;
        struct Node *node = head->next;

        // Iterate through the list and delete alternate nodes
        while (prev != NULL && node != NULL) {
            // Point the previous node to the next next node
            prev->next = node->next;

            // Free the memory of the current node
            free(node);

            // Update the previous and node pointers
            prev = prev->next;
            if (prev != NULL)
                node = prev->next;
        }
    }
};

File: 13-10-2024_Delete_Alternate_Nodes.java
class Solution {
    public void deleteAlt(Node head) {
        if (head == null) return;

        Node prev = head;
        Node node = head.next;

        while (prev != null && node != null) {
            prev.next = node.next;
            node = prev.next;
            if (node != null) {
                prev = node;
                node = node.next;
            }
        }
    }
}
